File: app/src/main/java/com/genius/wasylews/readme/di/component/AppComponent.java
package com.genius.wasylews.readme.di.component;

import com.genius.wasylews.readme.App;
import com.genius.wasylews.readme.di.module.AppModule;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
})
public interface AppComponent extends AndroidInjector<App> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<App> {
    }
}

File: app/src/main/java/com/genius/wasylews/readme/di/component/AppComponent.kt
package com.genius.wasylews.readme.di.component

import com.genius.wasylews.readme.App
import com.genius.wasylews.readme.di.module.AppModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [AndroidInjectionModule::class, AndroidSupportInjectionModule::class, AppModule::class])
interface AppComponent: AndroidInjector<App> {

    @Component.Factory
    interface Factory: AndroidInjector.Factory<App> {
        override fun create(@BindsInstance instance: App): AppComponent
    }
}